### ISSUE TITLE
ASPP software survey 2024

### ISSUE THREAD

**Issue by Debilski** [ASPP/installation#25]
This is the list of software packages that we install in https://github.com/ASPP/installation/blob/main/roles/common/tasks/packages.yml and in https://github.com/ASPP/installation/blob/main/roles/user-setup/tasks/main.yml 

Please check the box for any software that you required on the student laptops during ASPP 2024.

# system packages
- [ ] acl
- [ ] sudo
- [ ] plocate
- [ ] zram-generator-defaults
- [ ] diffutils
- [ ] gcc
- [ ] g++
- [ ] make
- [ ] wget
- [ ] curl
- [ ] gdb
- [ ] strace
- [ ] ltrace
- [ ] fpaste
- [ ] rsync
- [ ] man-db
- [ ] dstat
- [ ] openssh-clients
- [ ] openssh-server
- [ ] lz4
- [ ] git
- [ ] gitg
- [ ] gitk
- [ ] git-core-doc
- [ ] git-gui
- [ ] tig
- [ ] gprof2dot
- [ ] emacs
- [ ] vim-enhanced
- [ ] neovim
- [ ] python3-neovim
- [ ] fzf
- [ ] fd-find
- [ ] htop
- [ ] curl
- [ ] ncdu
- [ ] micro
- [ ] meson
- [ ] moreutils
- [ ] curl
- [ ] python3-Cython
- [ ] gnupg2
- [ ] hwloc
- [ ] tmux
- [ ] bash-completion
- [ ] bind-utils
- [ ] python3-ara
- [ ] python3-numpy
- [ ] python3-scipy
- [ ] python3-blosc
- [ ] python3-zmq
- [ ] python3-yaml
- [ ] python3-colorama
- [ ] python3-networkx
- [ ] python3-jupyter-client
- [ ] python3-jupyter-console
- [ ] python3-jupyter-c-kernel
- [ ] python3-jupyter-sphinx
- [ ] python3-ipython
- [ ] python3-pytest
- [ ] python3-dask
- [ ] python3-dask+array
- [ ] python3-dask+bag
- [ ] python3-dask+dataframe
- [ ] python3-dask+delayed
- [ ] python3-GitPython
- [ ] python3-graphviz
- [ ] python3-coverage
- [ ] python3-pip
- [ ] python3-plotly
- [ ] python3-opencv
- [ ] python3-toolz
- [ ] python3-joblib
- [ ] python3-scikit-learn
- [ ] python3-scipy
- [ ] python3-matplotlib
- [ ] python3-pyaudio
- [ ] python3-pyflakes
- [ ] python3-sphinx
- [ ] python3-sphinx_rtd_theme
- [ ] python3-pudb
- [ ] python3-pytest
- [ ] python3-flake8
- [ ] python3-pytest-flakes
- [ ] python3-pytest-mpi
- [ ] python3-pytest-cov
- [ ] python3-mpi4py-mpich
- [ ] python3-mpi4py-openmpi
- [ ] python3-imageio
- [ ] python3-h5py
- [ ] python3-scikit-image
- [ ] python3-xlrd
- [ ] python3-seaborn
- [ ] python3-psutil
- [ ] flexiblas-devel
- [ ] flite
- [ ] pyinstrument
- [ ] python3-pandas
- [ ] python3-bokeh
- [ ] python3-pysnooper
- [ ] python3-ipyparallel
- [ ] xauth
- [ ] dconf
- [ ] dbus-daemon
- [ ] mpv
- [ ] alsa-utils
- [ ] python3-systemd
- [ ] inkscape
- [ ] geeqie
- [ ] graphviz
- [ ] graphviz-doc
- [ ] openssh-askpass
- [ ] xclip
- [ ] firefox
- [ ] kcachegrind
- [ ] gedit
- [ ] gedit-plugins
- [ ] rxvt-unicode
- [ ] geany

# pip packages

- [ ] magic-wormhole
- [ ] ipython_memwatcher
- [ ] pyprof2calltree
- [ ] git+https://github.com/ASPP/prof3to2.git
- [ ] line-profiler
- [ ] numba
- [ ] altair
- [ ] snakeviz
- [ ] vega_datasets
- [ ] vega
- [ ] pelita==2.5.0
- [ ] memray

**Comment by keszybz** [comment]
I think this is a bit misleading. A lot of the stuff is enumerated because it forms the basic system (`acl`, `sudo`, `zram-generator-defaults`, `xclip`, `firefox`, …). Our tutors do not explicitly "use" any of those, but they are still needed.

**Comment by Debilski** [comment]
Yes, I was a bit lazy and did not filter them from the list in advance. But of course, we would keep these packages.

**Comment by otizonaizit** [comment]
curl is listed at least 3 times :)
